main.c

```diff
@@ -191,8 +193,68 @@ static cairo_status_t write_func(void *data, const unsigned char *buf,
 	return CAIRO_STATUS_SUCCESS;
 }
 
+void default_filename(char *filename, size_t n, int filetype) {
+	time_t time_epoch = time(NULL);
+	struct tm *time = localtime(&time_epoch);
+	if (time == NULL) {
+		perror("localtime");
+		exit(EXIT_FAILURE);
+	}
+
+	char *format_str;
+	switch (filetype) {
+	case GRIM_FILETYPE_PNG:
+		format_str = "%Y%m%d_%Hh%Mm%Ss_grim.png";
+		break;
+	case GRIM_FILETYPE_JPEG:
+#if HAVE_JPEG
+		format_str = "%Y%m%d_%Hh%Mm%Ss_grim.jpeg";
+		break;
+#else
+		assert(false);
+#endif
+	}
+
+	if (strftime(filename, n, format_str, time) == 0) {
+		fprintf(stderr, "failed to format datetime with strftime(3)\n");
+		exit(EXIT_FAILURE);
+	}
+}
+
+static bool path_exists(const char *path) {
+	return path && access(path, R_OK) != -1;
+}
+
+char *get_output_dir(void) {
+	static const char *output_dirs[] = {
+		"$GRIM_DEFAULT_DIR",
+		"$XDG_PICTURES_DIR",
+	};
+
+	wordexp_t p;
+	for (size_t i = 0; i < sizeof(output_dirs) / sizeof(char *); ++i) {
+		if (wordexp(output_dirs[i], &p, 0) == 0) {
```

Comment:
We can just use `getenv` instead of `wordexp` here. This would allow us to simplify this function and avoid the need to free `path`